src/components/Send.jsx

```diff
@@ -1,17 +1,104 @@
-import React from 'react'
+import React, { useEffect } from 'react'
 import { useState } from 'react'
 import { useLocation } from 'react-router-dom'
 import * as rb from 'react-bootstrap'
-import { serialize, ACCOUNTS } from '../utils'
+import PageTitle from './PageTitle'
+import ToggleSwitch from './ToggleSwitch'
+import { useCurrentWalletInfo, useSetCurrentWalletInfo, useCurrentWallet } from '../context/WalletContext'
+import { useSettings } from '../context/SettingsContext'
+import { serialize } from '../utils'
 import * as Api from '../libs/JmWalletApi'
 
-export default function Payment({ currentWallet }) {
+const CollaboratorsSelector = ({ numCollaborators, setNumCollaborators }) => {
+  const settings = useSettings()
+  const [usesCustomNumCollaborators, setUsesCustomNumCollaborators] = useState(false)
+
+  const collaboratorsSelection = [3, 5, 6, 9]
+
+  return (
+    <rb.Form.Group className="collaborators-selector">
+      <rb.Form.Label className="mb-0">Choose a number of collaborators ({numCollaborators})</rb.Form.Label>
+      <div className="mb-2">
+        <rb.Form.Text className="text-secondary">
+          A higher number is better for privacy, but also increases the fee.
+        </rb.Form.Text>
+      </div>
+      <div className="d-flex flex-row flex-wrap">
+        {collaboratorsSelection.map((number, index) => {
+          return (
+            <rb.Button
+              key={index}
+              variant="white"
+              className={`py-2 px-2 border rounded text-center${
+                !usesCustomNumCollaborators && numCollaborators === number ? ' border-dark border-2' : ''
+              }`}
+              onClick={() => {
+                setUsesCustomNumCollaborators(false)
+                setNumCollaborators(number)
+              }}
+            >
+              {number}
+            </rb.Button>
+          )
+        })}
+        <rb.Form.Control
+          type="number"
+          min={1}
+          placeholder="Other"
+          className={`py-2 px-2 border rounded text-center${usesCustomNumCollaborators ? ' border-dark border-2' : ''}`}
+          onChange={(e) => {
+            setUsesCustomNumCollaborators(true)
+            setNumCollaborators(e.target.value)
+          }}
+          onClick={(e) => {
+            if (e.target.value) {
+              setUsesCustomNumCollaborators(true)
+              setNumCollaborators(e.target.value)
+            }
+          }}
+        />
+      </div>
+      <rb.Form.Control.Feedback type="invalid">Please set the counterparties.</rb.Form.Control.Feedback>
+    </rb.Form.Group>
+  )
+}
+
+export default function Send({ currentWallet }) {
```

Comment:
How exactly does this `currentWallet` behave now compared to the `wallet` from `useCurrentWallet()`?

Reply:
Oh good catch, that's a remainder from the old version where the wallet was passed in as a prop. Removed.